Now you should run one of the following depending on your shell
source /share/apps/python/anaconda3.2019.3/etc/profile.d/conda.sh
source /share/apps/python/anaconda3.2019.3/etc/profile.d/conda.csh
2021-05-20 11:20:24.187617: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2021-05-20 11:20:36.791854: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set
2021-05-20 11:20:36.801935: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1
2021-05-20 11:20:37.139738: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
pciBusID: 0000:07:00.0 name: A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2021-05-20 11:20:37.141953: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 1 with properties: 
pciBusID: 0000:0f:00.0 name: A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2021-05-20 11:20:37.144118: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 2 with properties: 
pciBusID: 0000:47:00.0 name: A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2021-05-20 11:20:37.144139: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2021-05-20 11:20:37.787276: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.11
2021-05-20 11:20:37.787314: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.11
2021-05-20 11:20:38.102840: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10
2021-05-20 11:20:38.662424: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10
2021-05-20 11:20:39.089521: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.10
2021-05-20 11:20:39.290845: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.11
2021-05-20 11:20:39.456654: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.8
2021-05-20 11:20:39.470186: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0, 1, 2
2021-05-20 11:20:39.471393: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-05-20 11:20:39.486168: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
2021-05-20 11:20:39.832934: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
pciBusID: 0000:07:00.0 name: A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2021-05-20 11:20:39.835324: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 1 with properties: 
pciBusID: 0000:0f:00.0 name: A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2021-05-20 11:20:39.837451: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 2 with properties: 
pciBusID: 0000:47:00.0 name: A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2021-05-20 11:20:39.837486: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2021-05-20 11:20:39.837518: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.11
2021-05-20 11:20:39.837527: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.11
2021-05-20 11:20:39.837536: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10
2021-05-20 11:20:39.837545: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10
2021-05-20 11:20:39.837552: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.10
2021-05-20 11:20:39.837560: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.11
2021-05-20 11:20:39.837569: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.8
2021-05-20 11:20:39.850253: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0, 1, 2
2021-05-20 11:20:39.850282: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2021-05-20 11:20:43.089358: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-05-20 11:20:43.089427: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      0 1 2 
2021-05-20 11:20:43.089435: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 0:   N Y Y 
2021-05-20 11:20:43.089439: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 1:   Y N Y 
2021-05-20 11:20:43.089443: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 2:   Y Y N 
2021-05-20 11:20:43.101060: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 37571 MB memory) -> physical GPU (device: 0, name: A100-SXM4-40GB, pci bus id: 0000:07:00.0, compute capability: 8.0)
2021-05-20 11:20:43.105777: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 37571 MB memory) -> physical GPU (device: 1, name: A100-SXM4-40GB, pci bus id: 0000:0f:00.0, compute capability: 8.0)
2021-05-20 11:20:43.110301: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:2 with 37571 MB memory) -> physical GPU (device: 2, name: A100-SXM4-40GB, pci bus id: 0000:47:00.0, compute capability: 8.0)
2021-05-20 11:20:43.496488: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
2021-05-20 11:20:43.497238: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2245805000 Hz
2021-05-20 11:20:45.948919: I tensorflow/core/profiler/lib/profiler_session.cc:136] Profiler session initializing.
2021-05-20 11:20:45.948957: I tensorflow/core/profiler/lib/profiler_session.cc:155] Profiler session started.
2021-05-20 11:20:45.948990: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1365] Profiler found 3 GPUs
2021-05-20 11:20:45.958810: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcupti.so.11.0'; dlerror: libcupti.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /share/apps/cuda/11.0/lib:/share/apps/cuda/11.0/lib64:/usr/lib64/:/share/apps/python/anaconda3.2019.3/lib:/share/apps/cuda/11.0/lib64/stubs
2021-05-20 11:20:46.013291: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcupti.so
2021-05-20 11:20:46.356171: I tensorflow/core/profiler/lib/profiler_session.cc:172] Profiler session tear down.
2021-05-20 11:20:46.356338: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1487] CUPTI activity buffer flushed
2021-05-20 11:20:47.542369: W tensorflow/core/grappler/optimizers/data/auto_shard.cc:656] In AUTO-mode, and switching to DATA-based sharding, instead of FILE-based sharding as we cannot find appropriate reader dataset op(s) to shard. Error: Did not find a shardable source, walked to a node which is not a dataset: name: "FlatMapDataset/_8"
op: "FlatMapDataset"
input: "ParallelMapDatasetV2/_7"
attr {
  key: "Targuments"
  value {
    list {
    }
  }
}
attr {
  key: "f"
  value {
    func {
      name: "__inference_Dataset_flat_map_lambda_195"
    }
  }
}
attr {
  key: "output_shapes"
  value {
    list {
      shape {
        dim {
          size: 680
        }
      }
    }
  }
}
attr {
  key: "output_types"
  value {
    list {
      type: DT_FLOAT
    }
  }
}
. Consider either turning off auto-sharding or switching the auto_shard_policy to DATA to shard this dataset. You can do this by creating a new `tf.data.Options()` object then setting `options.experimental_distribute.auto_shard_policy = AutoShardPolicy.DATA` before applying the options object to the dataset via `dataset.with_options(options)`.
2021-05-20 11:20:51.797036: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.11
2021-05-20 11:20:53.798369: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.11
2021-05-20 11:21:03.970887: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:177] Filling up shuffle buffer (this may take a while): 53 of 131
2021-05-20 11:21:13.849850: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:177] Filling up shuffle buffer (this may take a while): 113 of 131
2021-05-20 11:21:16.768273: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:230] Shuffle buffer filled.
2021-05-20 11:21:16.773392: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
2021-05-20 11:21:18.225441: I tensorflow/core/profiler/lib/profiler_session.cc:136] Profiler session initializing.
2021-05-20 11:21:18.225483: I tensorflow/core/profiler/lib/profiler_session.cc:155] Profiler session started.
WARNING:tensorflow:Callback method `on_train_batch_begin` is slow compared to the batch time (batch time: 0.0104s vs `on_train_batch_begin` time: 0.1496s). Check your callbacks.
WARNING:tensorflow:Callback method `on_train_batch_end` is slow compared to the batch time (batch time: 0.0104s vs `on_train_batch_end` time: 0.0107s). Check your callbacks.
2021-05-20 11:21:19.281972: I tensorflow/core/profiler/lib/profiler_session.cc:71] Profiler session collecting data.
2021-05-20 11:21:19.298436: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1487] CUPTI activity buffer flushed
2021-05-20 11:21:19.510562: I tensorflow/core/profiler/internal/gpu/cupti_collector.cc:228]  GpuTracer has collected 7257 callback api events and 7228 activity events. 
2021-05-20 11:21:19.712868: I tensorflow/core/profiler/lib/profiler_session.cc:172] Profiler session tear down.
2021-05-20 11:21:19.900330: I tensorflow/core/profiler/rpc/client/save_profile.cc:137] Creating directory: /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19
2021-05-20 11:21:20.072193: I tensorflow/core/profiler/rpc/client/save_profile.cc:143] Dumped gzipped tool data for trace.json.gz to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.trace.json.gz
2021-05-20 11:21:20.218366: I tensorflow/core/profiler/rpc/client/save_profile.cc:137] Creating directory: /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19
2021-05-20 11:21:20.244115: I tensorflow/core/profiler/rpc/client/save_profile.cc:143] Dumped gzipped tool data for memory_profile.json.gz to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.memory_profile.json.gz
2021-05-20 11:21:20.339217: I tensorflow/core/profiler/rpc/client/capture_profile.cc:251] Creating directory: /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19Dumped tool data for xplane.pb to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.xplane.pb
Dumped tool data for overview_page.pb to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.overview_page.pb
Dumped tool data for input_pipeline.pb to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.input_pipeline.pb
Dumped tool data for tensorflow_stats.pb to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.tensorflow_stats.pb
Dumped tool data for kernel_stats.pb to /qfs/people/jain432/pacer/logs/fit_v10/e50_TFDataOptMGPU_g3_20210520-112036/train/plugins/profile/2021_05_20_11_21_19/a100-01.kernel_stats.pb

